akka-docs-dev/rst/scala/code/docs/stream/FlowDocSpec.scala

```diff
@@ -142,17 +143,17 @@ class FlowDocSpec extends AkkaSpec {
   "various ways of transforming materialized values" in {
     import scala.concurrent.duration._
 
-    val throttler = Flow(Source(1.second, 1.second, "test")) { implicit builder =>
+    val throttler = Flow.fromGraph(FlowGraph.create(Source(1.second, 1.second, "test")) { implicit builder =>
       tickSource =>
         import FlowGraph.Implicits._
         val zip = builder.add(ZipWith[String, Int, Int](Keep.right))
         tickSource ~> zip.in0
-        (zip.in1, zip.out)
-    }
+        FlowShape(zip.in1, zip.out)
+    })
 
     //#flow-mat-combine
-    // An empty source that can be shut down explicitly from the outside
-    val source: Source[Int, Promise[Unit]] = Source.lazyEmpty[Int]
+    // An source that can be signalled explicitly from the outside
+    val source: Source[Int, Promise[Option[Int]]] = Source.maybe[Int]
```

Comment:
Hm, I think we should replace this with a tick source, that materializes to a Cancellable. Might be better for this doc example as it is more intuitive to get what the Cancellable does compared to a Promise[Option[Int]]